plugins/modules/host.py

```diff
@@ -197,11 +353,44 @@
 
 from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import (
     ensure_puppetclasses,
+    interfaces_foreman_spec, interfaces_ansible_spec,
     ForemanEntityAnsibleModule,
     HostMixin,
 )
 
 
+def ensure_host_interfaces(module, entity, interfaces):
+    scope = {'host_id': entity['id']}
+
+    current_interfaces = module.list_resource('interfaces', params=scope)
+    current_interfaces_ids = {x['id'] for x in current_interfaces}
+    expected_interfaces_ids = set()
+
+    for interface in interfaces:
+        for possible_identifier in ['identifier', 'name']:
+            if possible_identifier in interface:
+                unique_identifier = possible_identifier
+                break
+        else:
+            unique_identifier = None
+            # TODO: this shouldn't happen
+
+        if 'mac' in interface:
+            interface['mac'] = interface['mac'].lower()
+
+        existing_interface = next((x for x in current_interfaces if x.get(unique_identifier) == interface[unique_identifier]), None)
```

Comment:
```suggestion
        existing_interface = next((x for x in current_interfaces if unique_identifier and x.get(unique_identifier) == interface[unique_identifier]), None)
```

Reply:
Yeah, so I needed to "find" an existing interface somehow to know if we're updating it or creating a new one. My idea was to look if there is an interface defined with the same identifier, hostname or mac (and I didn't add mac in my code, ups). This behavior is not documented yet.

Reply:
I think it would make sense for PXE deployment (`mac` is probably the best option since it is required) but for image based provisioning the changes won't be propagated to the VM since most compute resources does not manage host update (afaik, same applies to `compute_attributes`). Shouldn't we just exclude `interfaces_attributes` in this case?

Reply:
Is there *anything* that we could ask the user to provide to make the entry unique? If you for example use identifier:fam01 (like I did in the tests), would the interface be properly deployed with that name inside the OS, even if we don't define ip/mac to match on?

Reply:
So I had a few more thoughts:
* I'll look what hammer does tomorrow
* If both the user and API both provide exactly one interface, that's the one we're updating, no need to match
* In the case we couldn't find an unique identifier, proceed to add a new interface and warn the user that this is probably not what they want

Reply:
Sounds good! We just need to be careful with Foreman setting `ignore_puppet_facts_for_provisioning`, disabled by default. Puppet facts may update some values and/or add new interfaces. If new interfaces are created by puppet facts, we need to match, and if the identifier has changed, we might create a new interface at each run, which is not what we want :)

Reply:
I've implemented the "only one" and "warning" proposals from above.

Now to hammer: 
* Given the non-declarative way hammer is designed, you have individual commands to "add", "update" and "delete" host interfaces.
* Let's ignore adding, that is trivial.
* Deleting an interface requires two inputs: host id (or name) and interface id (which you need to look up manually via `hammer host interfaces list` before)
* Updating also (technically) requires host id/name and interface id, but nothing *enforces* the entry of the ID, and if you don't you get the good old "Interfaces some interfaces are invalid" error because it tries to add an interface with an identifier that's already present.

So, all in all, I'm no way wiser now.